paddle/fluid/framework/parallel_executor.cc

```diff
@@ -216,24 +224,29 @@ class ParallelExecutorPrivate {
     const std::string var_name = "NCCLCommunicator";
     auto var = scope->FindVar(var_name);
     if (var != nullptr) {
-      PADDLE_ENFORCE(var->IsInitialized(),
-                     "if %s exists, it must be initialized", var_name);
+      PADDLE_ENFORCE_EQ(var->IsInitialized(), true,
+                        platform::errors::PreconditionNotMet(
+                            "if %s exists, it must be initialized", var_name));
       VLOG(1) << "find " << var_name
               << " in scope, so use it and does not recreate!";
       nccl_ctxs_ = var->GetMutable<platform::NCCLCommunicator>();
       return;
     }
 
     if (bst->use_hierarchical_allreduce_) {
-      PADDLE_ENFORCE(bst->num_trainers_ > 1, "num_trainers:%llu < 1",
-                     bst->num_trainers_);
-      PADDLE_ENFORCE(bst->hierarchical_allreduce_inter_nranks_ > 1,
-                     "inter_nranks:%d < 1",
-                     bst->hierarchical_allreduce_inter_nranks_);
-      PADDLE_ENFORCE(
-          (bst->num_trainers_ % bst->hierarchical_allreduce_inter_nranks_ == 0),
-          "num_trainers:%llu mod inter_nranks:%d != 0", bst->num_trainers_,
-          bst->hierarchical_allreduce_inter_nranks_);
+      PADDLE_ENFORCE_GT(bst->num_trainers_, 1,
+                        platform::errors::PreconditionNotMet(
+                            "num_trainers is %llu, which is less than 1",
+                            bst->num_trainers_));
+      PADDLE_ENFORCE_GT(bst->hierarchical_allreduce_inter_nranks_, 1,
+                        platform::errors::PreconditionNotMet(
+                            "inter_nranks is %d, which is less than 1",
+                            bst->hierarchical_allreduce_inter_nranks_));
```

Comment:
`"The inter_nranks should be greater than 1, but received %d.", bst->hierarchical_allreduce_inter_nranks_`

Reply:
Done